refactor(account-management)_: improvement on consistent method signatures

saledjenic opened the pull request:
- Eliminated the main wallet account address from the account manager.
- Introduced mutexes for thread-safe access to keystore and selected chat account.
- Updated code affected by the removal.
- Updated related tests to reflect the removal.
- Updated account selection and validation methods to utilize the new account structure.
- Refactored tests to align with changes in account management.
- Removed deprecated methods and cleaned up unused code to enhance maintainability.
- Improved clarity in account-related operations by adopting consistent method signatures and return types.

----
Reference to other accounts-management refactor PRs:
- https://github.com/status-im/status-go/pull/6605
- https://github.com/status-im/status-go/pull/6612
- https://github.com/status-im/status-go/pull/6640
- https://github.com/status-im/status-go/pull/6660
- https://github.com/status-im/status-go/pull/6668
- https://github.com/status-im/status-go/pull/6673
- https://github.com/status-im/status-go/pull/6686
- https://github.com/status-im/status-go/pull/6687
- https://github.com/status-im/status-go/pull/6692

saledjenic committed:
refactor(account-management)_: remove main wallet account address handling from account manager

- Eliminated the main wallet account address from the account manager.
- Introduced mutexes for thread-safe access to keystore and selected chat account.
- Updated code affected by the removal.
- Updated related tests to reflect the removal.

saledjenic committed:
refactor(account-management)_: improvement on consistent method signatures

- Updated account selection and validation methods to utilize the new account structure.
- Refactored tests to align with changes in account management.
- Removed deprecated methods and cleaned up unused code to enhance maintainability.
- Improved clarity in account-related operations by adopting consistent method signatures and return types.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (103)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/1/) | 2025-06-16 14:43:36 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250616-144047-dda316-pr6686.aar) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/1/) | 2025-06-16 14:43:58 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-144053-dda316-pr6686.zip) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/1/) | 2025-06-16 14:44:01 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-144049-dda316-pr6686.zip) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/1/) | 2025-06-16 14:45:30 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250616-144055-dda316-pr6686.zip) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/1/) | 2025-06-16 14:45:37 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250616-144100-dda316-pr6686.zip) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/1/) | 2025-06-16 14:48:04 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250616-144048-dda316-pr6686.zip) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/1/) | 2025-06-16 14:49:04 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/1//consoleText) |
| :heavy_check_mark: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/1/) | 2025-06-16 14:50:52 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250616-144050-dda316-pr6686.zip) |
| :heavy_multiplication_x: | dda31628 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/1/) | 2025-06-16 15:06:18 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/2/) | 2025-06-17 06:30:08 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-062718-210787-pr6686.aar) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/2/) | 2025-06-17 06:30:33 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-062724-210787-pr6686.zip) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/2/) | 2025-06-17 06:30:53 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-062721-210787-pr6686.zip) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/2/) | 2025-06-17 06:32:00 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-062726-210787-pr6686.zip) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/2/) | 2025-06-17 06:32:10 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-062730-210787-pr6686.zip) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/2/) | 2025-06-17 06:34:58 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/2//consoleText) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/2/) | 2025-06-17 06:35:08 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-062718-210787-pr6686.zip) |
| :heavy_check_mark: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/2/) | 2025-06-17 06:37:08 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-062722-210787-pr6686.zip) |
| :heavy_multiplication_x: | 21078772 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/2/) | 2025-06-17 06:52:47 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/3/) | 2025-06-17 10:03:43 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-095921-de61ba-pr6686.zip) |
| :heavy_check_mark: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/3/) | 2025-06-17 10:04:04 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-095914-de61ba-pr6686.zip) |
| :x: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/3/) | 2025-06-17 10:04:18 | ~4 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/3/consoleText) |
| :heavy_check_mark: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/3/) | 2025-06-17 10:05:14 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-095919-de61ba-pr6686.zip) |
| :heavy_check_mark: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/3/) | 2025-06-17 10:05:56 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-095913-de61ba-pr6686.aar) |
| :heavy_check_mark: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/3/) | 2025-06-17 10:09:00 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-095915-de61ba-pr6686.zip) |
| :heavy_check_mark: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/3/) | 2025-06-17 10:13:19 | ~14 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-095915-de61ba-pr6686.zip) |
| :heavy_multiplication_x: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/3/) | 2025-06-17 10:17:05 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/3//consoleText) |
| :heavy_multiplication_x: | de61ba1b | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/3/) | 2025-06-17 10:34:24 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/4/) | 2025-06-17 11:30:52 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-112730-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/4/) | 2025-06-17 11:31:53 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-112733-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/4/) | 2025-06-17 11:31:53 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-112723-e0f721-pr6686.aar) |
| :x: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/4/) | 2025-06-17 11:32:06 | ~4 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/4/consoleText) |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/4/) | 2025-06-17 11:32:47 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-112725-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/4/) | 2025-06-17 11:32:55 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-112726-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/4/) | 2025-06-17 11:37:49 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/4//consoleText) |
| :heavy_check_mark: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/4/) | 2025-06-17 11:40:59 | ~13 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-112727-e0f721-pr6686.zip) |
| :heavy_multiplication_x: | e0f7215a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/4/) | 2025-06-17 11:56:20 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/4//consoleText) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/5/) | 2025-06-17 12:16:03 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-121315-e0f721-pr6686.aar) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/5/) | 2025-06-17 12:16:27 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-121317-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/5/) | 2025-06-17 12:17:14 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121320-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/5/) | 2025-06-17 12:18:35 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-121327-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/5/) | 2025-06-17 12:18:46 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121324-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/5/) | 2025-06-17 12:19:57 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/5//consoleText) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/5/) | 2025-06-17 12:20:55 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-121316-e0f721-pr6686.zip) |
| :heavy_check_mark: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/5/) | 2025-06-17 12:22:16 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-121318-e0f721-pr6686.zip) |
| :heavy_multiplication_x: | e0f7215a | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/5/) | 2025-06-17 12:40:10 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/6/) | 2025-06-17 12:23:20 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250617-121838-3ecce4-pr6686.zip) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/6/) | 2025-06-17 12:24:10 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121852-3ecce4-pr6686.zip) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/6/) | 2025-06-17 12:24:15 | ~8 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250617-121610-3ecce4-pr6686.aar) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/6/) | 2025-06-17 12:24:58 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250617-121725-3ecce4-pr6686.zip) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/6/) | 2025-06-17 12:25:23 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-121631-3ecce4-pr6686.zip) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/6/) | 2025-06-17 12:28:57 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250617-122111-3ecce4-pr6686.zip) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/6/) | 2025-06-17 12:42:41 | ~22 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/6//consoleText) |
| :heavy_check_mark: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/6/) | 2025-06-17 12:46:15 | ~23 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250617-122220-3ecce4-pr6686.zip) |
| :heavy_multiplication_x: | 3ecce4cf | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/6/) | 2025-06-17 13:11:49 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/6//consoleText) |
| :heavy_multiplication_x: | 3ecce4cf | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/7/) | 2025-06-18 06:29:46 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/7//consoleText) |
| :heavy_multiplication_x: | 3ecce4cf | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/8/) | 2025-06-18 10:36:44 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/7/) | 2025-06-18 13:09:57 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-130611-fc87a5-pr6686.zip) |
| :heavy_check_mark: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/7/) | 2025-06-18 13:10:40 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250618-130605-fc87a5-pr6686.aar) |
| :heavy_check_mark: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/7/) | 2025-06-18 13:10:45 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-130614-fc87a5-pr6686.zip) |
| :x: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/7/) | 2025-06-18 13:11:15 | ~4 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/7/consoleText) |
| :heavy_check_mark: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/7/) | 2025-06-18 13:11:16 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250618-130606-fc87a5-pr6686.zip) |
| :heavy_check_mark: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/7/) | 2025-06-18 13:13:53 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-130607-fc87a5-pr6686.zip) |
| :heavy_multiplication_x: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/7/) | 2025-06-18 13:20:43 | ~14 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/7//consoleText) |
| :heavy_check_mark: | fc87a5d6 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/7/) | 2025-06-18 13:23:21 | ~17 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-130609-fc87a5-pr6686.zip) |
| :heavy_multiplication_x: | fc87a5d6 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/9/) | 2025-06-18 13:39:19 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/9//consoleText) |
| :heavy_multiplication_x: | fc87a5d6 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/10/) | 2025-06-18 15:21:37 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/8/) | 2025-06-18 16:11:14 | ~2 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-160822-8336aa-pr6686.zip) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/8/) | 2025-06-18 16:13:27 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250618-160829-8336aa-pr6686.zip) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/8/) | 2025-06-18 16:14:19 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250618-160825-8336aa-pr6686.zip) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/8/) | 2025-06-18 16:14:44 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250618-160817-8336aa-pr6686.aar) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/8/) | 2025-06-18 16:15:10 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-160818-8336aa-pr6686.zip) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/8/) | 2025-06-18 16:15:14 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250618-160818-8336aa-pr6686.zip) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/8/) | 2025-06-18 16:22:53 | ~14 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/8//consoleText) |
| :heavy_check_mark: | 8336aab1 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/8/) | 2025-06-18 16:23:23 | ~15 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250618-160819-8336aa-pr6686.zip) |
| :heavy_multiplication_x: | 8336aab1 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/11/) | 2025-06-18 16:43:19 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/9/) | 2025-06-19 12:03:19 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250619-120040-5db552-pr6686.aar) |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/9/) | 2025-06-19 12:03:50 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-120046-5db552-pr6686.zip) |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/9/) | 2025-06-19 12:04:29 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-120042-5db552-pr6686.zip) |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/9/) | 2025-06-19 12:05:27 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-120049-5db552-pr6686.zip) |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/9/) | 2025-06-19 12:06:21 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250619-120042-5db552-pr6686.zip) |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/9/) | 2025-06-19 12:06:30 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250619-120052-5db552-pr6686.zip) |
| :heavy_multiplication_x: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/9/) | 2025-06-19 12:11:01 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/9//consoleText) |
| :heavy_check_mark: | 5db55208 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/9/) | 2025-06-19 12:12:03 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-120044-5db552-pr6686.zip) |
| :heavy_multiplication_x: | 5db55208 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/12/) | 2025-06-19 12:31:02 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/12//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/10/) | 2025-06-19 13:20:59 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250619-131838-b3979d-pr6686.aar) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/10/) | 2025-06-19 13:22:18 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-131845-b3979d-pr6686.zip) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/10/) | 2025-06-19 13:22:30 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-131840-b3979d-pr6686.zip) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/10/) | 2025-06-19 13:23:30 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250619-131852-b3979d-pr6686.zip) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/10/) | 2025-06-19 13:25:07 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-131848-b3979d-pr6686.zip) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/10/) | 2025-06-19 13:25:09 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250619-131840-b3979d-pr6686.zip) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/10/) | 2025-06-19 13:27:16 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/10//consoleText) |
| :heavy_check_mark: | b3979d10 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/10/) | 2025-06-19 13:28:48 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-131842-b3979d-pr6686.zip) |
| :heavy_multiplication_x: | b3979d10 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/13/) | 2025-06-19 13:47:22 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/11/) | 2025-06-19 13:54:17 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250619-135152-aeecb4-pr6686.aar) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/11/) | 2025-06-19 13:55:16 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-135154-aeecb4-pr6686.zip) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/11/) | 2025-06-19 13:55:31 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-135157-aeecb4-pr6686.zip) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/11/) | 2025-06-19 13:56:56 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250619-135203-aeecb4-pr6686.zip) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/11/) | 2025-06-19 13:58:15 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250619-135153-aeecb4-pr6686.zip) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/11/) | 2025-06-19 13:58:42 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250619-135200-aeecb4-pr6686.zip) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/11/) | 2025-06-19 14:00:27 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/11//consoleText) |
| :heavy_check_mark: | aeecb46f | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/11/) | 2025-06-19 14:03:47 | ~11 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250619-135155-aeecb4-pr6686.zip) |
| :heavy_multiplication_x: | aeecb46f | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/14/) | 2025-06-19 14:29:52 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/14//consoleText) |
| :heavy_check_mark: | aeecb46f | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/15/) | 2025-06-19 14:57:24 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/15//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/12/) | 2025-06-20 07:18:34 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250620-071528-70a5c4-pr6686.aar) |
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/12/) | 2025-06-20 07:19:18 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-071530-70a5c4-pr6686.zip) |
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/12/) | 2025-06-20 07:19:35 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-071533-70a5c4-pr6686.zip) |
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6686/12/) | 2025-06-20 07:20:41 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250620-071540-70a5c4-pr6686.zip) |
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/12/) | 2025-06-20 07:20:47 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250620-071536-70a5c4-pr6686.zip) |
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/12/) | 2025-06-20 07:23:16 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250620-071529-70a5c4-pr6686.zip) |
| :heavy_multiplication_x: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/12/) | 2025-06-20 07:24:12 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/12//consoleText) |
| :heavy_check_mark: | 70a5c49b | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/12/) | 2025-06-20 07:28:16 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250620-071532-70a5c4-pr6686.zip) |
| :heavy_check_mark: | 70a5c49b | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/16/) | 2025-06-20 07:41:25 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/16//consoleText) |
| :heavy_multiplication_x: | 70a5c49b | [#13](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/13/) | 2025-06-23 08:26:26 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/13//consoleText) |
| :heavy_multiplication_x: | 70a5c49b | [#14](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/14/) | 2025-06-23 11:30:49 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/14//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8a66799f | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6686/13/) | 2025-06-23 20:00:24 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250623-195742-8a6679-pr6686.aar) |
| :heavy_check_mark: | 8a66799f | [#13](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6686/13/) | 2025-06-23 20:01:12 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250623-195748-8a6679-pr6686.zip) |
| :heavy_check_mark: | 8a66799f | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6686/13/) | 2025-06-23 20:01:31 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250623-195744-8a6679-pr6686.zip) |
| :heavy_check_mark: | 8a66799f | [#13](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6686/13/) | 2025-06-23 20:02:19 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250623-195750-8a6679-pr6686.zip) |
| :heavy_check_mark: | 8a66799f | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6686/13/) | 2025-06-23 20:03:19 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250623-195743-8a6679-pr6686.zip) |
| :heavy_multiplication_x: | 8a66799f | [#15](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/15/) | 2025-06-23 20:06:38 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6686/15//consoleText) |
| :x: | 8a66799f | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/13/) | 2025-06-23 20:07:36 | ~9 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6686/13/consoleText) |
| :heavy_check_mark: | 8a66799f | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/17/) | 2025-06-23 20:25:17 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6686/17//consoleText) |

saledjenic committed:
refactor(account-management)_: account package renamed to accountsmanagement package

saledjenic committed:
refactor(account-management)_: rename DefaultManager to AccountsManager and update references

saledjenic committed:
chore(accounts-management)_: add README for Accounts Management package

Added a detailed README.md file outlining the functionality, structure, key components, and usage examples for the Accounts Management package.

saledjenic committed:
refactor(accounts-management)_: rename account selection methods

- Renamed `SelectAccount` to `SetChatAccount`.
- Renamed `SetChatAccount` to `SetChatAccountWithPrivateKey`.
- Updated related test methods to reflect the new naming convention.
- Moved the `LoginParams` struct from the `types` package to `geth_backend` file.
- Adjusted README documentation to reflect changes in account selection methods.

alaibe commented on services/wallet/api.go:
```diff
@@ -826,30 +822,6 @@ func (api *API) FetchChainIDForURL(ctx context.Context, rpcURL string) (*big.Int
 	return client.ChainID(ctx)
 }
 
-func (api *API) getVerifiedWalletAccount(address, password string) (*accounttypes.SelectedExtKey, error) {
```
why is this removed?

saledjenic replied:
Look at the line `871`, this function is now part of accounts management, and does the same as here, but just moved there.

alaibe commented on services/personal/api.go:
```diff
@@ -3,7 +3,7 @@ package personal
 import (
```
is the personal api still use? should it not belong to account?

saledjenic replied:
Yes, the client is still using the personal API and we cannot remove it, or not yet. In one of the follow-up PRs, a sign function will become part of the accounts management package.

saledjenic committed:
refactor(accounts-management)_: move KeyStore interface from types package to accountsmanagement package

saledjenic committed:
refactor(account-management)_: centralize account path constants in accounts common package

saledjenic committed:
fix_: failing TestChangeDatabasePassword test fixed

Due to changes done around account management, TestChangeDatabasePassword started failing.
It turned out that the issues we had just popped up now, cause with the previous implementation they were masked.

What's fixed:
- different behaviour in test execution based on the platform, because of the way how MacOS resolves symlinks
in comparison to Linux OS
- `TestChangeDatabasePassword` improved, cause the previous implementation was just skipping some errors and checks
- improved keystore file location handling, cause we have to support both relative and absolute locations in order to be
backward compatible (due to old mobile users).

dlipicar commented on server/pairing/common.go:
```diff
@@ -49,7 +50,12 @@ func validateKeys(keys map[string][]byte, password string) error {
+			return err
+		}
+
+		err = acc.ValidateExtendedKey()
```
this check is done in `CreateAccountFromKey` already, isn't it?

saledjenic replied:
Correct. Will remove that check.

dlipicar commented on account/generator/types.go:
```diff
@@ -45,10 +51,16 @@ func (a *Account) PublicKey() *ecdsa.PublicKey {
 }
 
 func (a *Account) PublicKeyHex() string {
+	if a.privateKey == nil {
```
since the privateKey seems to always be derivable from the extKey, this would only be a case where the extKey passed to the constructor is nil as well, right? Is that something we want to allow? 
Perhaps we can move some checks done in `CreateAccountFromKey` (consistency and correctness of privateKey and extKey) here (instead of `NewAccount` have a `MakeAccount` returning an err parameter) to ensure every instance of Account is a valid one and avoid further checks down the line. Unless I'm missing some use case for an "invalid" account object.

saledjenic replied:
> since the privateKey seems to always be derivable from the extKey, this would only be a case where the extKey passed to the constructor is nil as well, right? 

Not only when `extKey`, but when both `privateKey` and `extKey` are `nil`. So there is no other way to create an account except using `NewAccount`, cause `Account`'s fields are private.
All of the following are correct to do in order to create account:
- `NewAccount(nil, nil)`
- `NewAccount(privKey, nil)` <- only priv key will be set
- `NewAccount(privKey, extKey)` <- both priv and ext keys will be set
- `NewAccount(nil, extKey)` <- both priv and ext keys will be set

This is the implementation:
```go
func NewAccount(privateKey *ecdsa.PrivateKey, extKey *extkeys.ExtendedKey) *Account {
	if privateKey == nil && extKey != nil {
		privateKey = extKey.ToECDSA()
	}

	return &Account{
		privateKey:  privateKey,
		extendedKey: extKey,
	}
}
```

> Is that something we want to allow?

Yes, cause we support creating accounts from private key only, in that case no way to figure out the extended key and that's why all functions deals with private key, cause it will be always there either you provide just a private key, just extended key or both to the constructor.

dlipicar commented on account/accounts.go:
```diff
@@ -30,11 +28,11 @@ func (e ErrCannotLocateKeyFile) Error() string {
 
-	selectedChatAccount *types.SelectedExtKey // account that was processed during the last call to SelectAccount()
-	mainAccountAddress  ethtypes.Address
+	selectedChatAccountMutex sync.RWMutex
```
This is kind of giving me the idea that the keystore and the selected account are not bound together (and so, can be protected by independent mutexes). Is that the case? Is it fine to change the keystore and retain the selected chat account? Or should the selected chat account always be invalidated when the keystore changes?

dlipicar replied:
(also managing multiple mutexes in the same struct is fine if needed, but normally gives me deadlock chills)

saledjenic replied:
Yes, the selected chat account determines the keystore that will be used, but I couldn't bind them together (at least not at this moment) cause the way how it's used sets them independently. That's for sure one thing we need to think of and improve in future improvements.

> also managing multiple mutexes in the same struct is fine if needed, but normally gives me deadlock chills

Yes, that's true, but the way they are used now allows those two mutexes.

saledjenic committed:
refactor(accounts-management)_: enhance AccountsManager with persistence interface

- Introduced a new `Persistence` interface to manage account data storage and retrieval.
- Updated `AccountsManager` to require a `Persistence` instance during initialization.
- Refactored related methods to utilize the new persistence layer, improving separation of concerns.
- Updated tests to mock the persistence layer for better isolation and reliability.
- Added error handling for missing persistence and logger instances.

saledjenic committed:
refactor(accounts-management)_: simplify MigrateKeyStoreDir logic

There is no need for the addresses to be provided from outside, the migration should migrate all the files
that are in the keystore dir to another location.

- Removed the addresses parameter from the MigrateKeyStoreDir method in AccountsManager and related interfaces.
- Updated all references and usages of MigrateKeyStoreDir across the codebase to reflect the new signature.
- Adjusted tests and documentation accordingly to ensure consistency and clarity.
